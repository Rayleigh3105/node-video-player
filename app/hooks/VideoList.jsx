import React from 'react';
import PropTypes from 'prop-types';

const getFriendlyName = (fileName) => {
  const parts = fileName.split('.');
  parts.splice(-1, 1);

  return parts.join(' ').replace('_', ' ');
};

const VideoList = ({ videos, onClick }) => (
  <div className="row">
    {
        videos.map((v) => (
          <div
            key={v}
            tabIndex={0}
            role="link"
            className="col-md-3 cursor-pointer text-center"
            onClick={() => onClick(v)}
            onKeyPress={() => onClick(v)}
          >
            <img
              alt={getFriendlyName(v)}
              src={`/videos/${v}.jpg`}
              className="thumbnail"
            />
            {getFriendlyName(v)}
          </div>
        ))
      }
  </div>
);

VideoList.propTypes = {
  onClick: PropTypes.func,
  videos: PropTypes.arrayOf(PropTypes.string).isRequired,
};

VideoList.defaultProps = {
  onClick: () => {},
};

export default VideoList;
