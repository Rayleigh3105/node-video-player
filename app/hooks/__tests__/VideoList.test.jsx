import { create, act } from 'react-test-renderer';
import React from 'react';
import VideoList from '../VideoList';

const createSubject = () => {
  let subject;
  const onClick = jest.fn();

  act(() => {
    subject = create(
      <VideoList onClick={onClick} videos={['file1.mp4', 'file2.mp4', 'file3.mp4']} />,
    );
  });

  return { subject, onClick };
};

it('should render correctly', () => {
  const { subject } = createSubject();
  expect(subject.toJSON()).toMatchSnapshot();
});

describe('when a video is clicked', () => {
  it('should invoke `props.onClick` with the video name', () => {
    const { subject, onClick } = createSubject();
    const videos = subject.root.findAllByProps({ role: 'link' });

    videos[0].props.onClick();
    expect(onClick).toHaveBeenCalledWith('file1.mp4');
  });
});

describe('when a video receives a key press', () => {
  it('should invoke `props.onClick` with the video name', () => {
    const { subject, onClick } = createSubject();
    const videos = subject.root.findAllByProps({ role: 'link' });

    videos[0].props.onKeyPress();
    expect(onClick).toHaveBeenCalledWith('file1.mp4');
  });
});