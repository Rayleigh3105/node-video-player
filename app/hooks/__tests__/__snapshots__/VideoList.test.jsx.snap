// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  className="row"
>
  <div
    className="col-md-3 cursor-pointer text-center"
    onClick={[Function]}
    onKeyPress={[Function]}
    role="link"
    tabIndex={0}
  >
    <img
      alt="file1"
      className="thumbnail"
      src="/videos/file1.mp4.jpg"
    />
    file1
  </div>
  <div
    className="col-md-3 cursor-pointer text-center"
    onClick={[Function]}
    onKeyPress={[Function]}
    role="link"
    tabIndex={0}
  >
    <img
      alt="file2"
      className="thumbnail"
      src="/videos/file2.mp4.jpg"
    />
    file2
  </div>
  <div
    className="col-md-3 cursor-pointer text-center"
    onClick={[Function]}
    onKeyPress={[Function]}
    role="link"
    tabIndex={0}
  >
    <img
      alt="file3"
      className="thumbnail"
      src="/videos/file3.mp4.jpg"
    />
    file3
  </div>
</div>
`;
